Persephone/Components/Queue/CurrentCoverArtView.swift

```diff
@@ -8,23 +8,39 @@
 
 import AppKit
 import ReSwift
+import Kingfisher
 
 class CurrentCoverArtView: NSImageView {
   required init?(coder: NSCoder) {
     super.init(coder: coder)
 
     App.store.subscribe(self) {
-      $0.select { $0.playerState.currentArtwork }
+      $0.select { $0.playerState.currentSong }
     }
   }
+
+  func setAlbumImage(_ album: Album) {
+    guard let imagePath = album.coverArtFilePath else { return }
+
+    let imageURL = URL(fileURLWithPath: imagePath)
+    let provider = LocalFileImageDataProvider(fileURL: imageURL)
+    self.kf.setImage(
+      with: .provider(provider),
+      placeholder: NSImage.defaultCoverArt,
+      options: [
+        .processor(DownsamplingImageProcessor(size: NSSize(width: 500, height: 500))),
+        .scaleFactor(2),
+      ]
+    )
+  }
 }
 
 extension CurrentCoverArtView: StoreSubscriber {
-  typealias StoreSubscriberStateType = NSImage?
+  typealias StoreSubscriberStateType = Song?
 
-  func newState(state: NSImage?) {
-    if let coverArt = state {
-      image = coverArt
+  func newState(state: Song?) {
+    if let song = state {
+      setAlbumImage(song.album)
```

Comment:
I could see this being a `guard let` just from a be-more-Swifty perspective, but at the same time, the gains would be minimal.